lua/precognition/init.lua

```diff
@@ -3,41 +3,73 @@ local vm = require("precognition.vertical_motions")
 
 local M = {}
 
----@alias SupportedHints "'^'" | "'b'" | "'w'" | "'$'"
----@alias SupportedGutterHints "G" | "gg" | "{" | "}"
+---@alias Dollar "$"
+---@alias PrevParagraph "{"
+---@alias NextParagraph "}"
+
+---@class HintOpts
+---@field text string
+---@field prio integer
+
+---@alias PlaceLoc integer
+
+---@class (exact) HintConfig
+---@field w HintOpts
+---@field e HintOpts
+---@field b HintOpts
+---@field Carat HintOpts
+---@field Dollar HintOpts
+
+---@class GutterHintConfig
+---@field G HintOpts
+---@field gg HintOpts
+---@field PrevParagraph HintOpts
+---@field NextParagraph HintOpts
 
 ---@class Precognition.Config
 ---@field startVisible boolean
----@field hints { SupportedHints: { text: string, prio: integer } }
----@field gutterHints { SupportedGutterHints: { text: string, prio: integer } }
+---@field hints HintConfig
+---@field showBlankVirtLine boolean
+---@field gutterHints GutterHintConfig
 
 ---@class Precognition.PartialConfig
 ---@field startVisible? boolean
----@field hints? { SupportedHints: { text: string, prio: integer } }
----@field gutterHints? { SupportedGutterHints: { text: string, prio: integer } }
-
----@alias Precognition.VirtLine {  SupportedHints: integer | nil }
-
----@alias Precognition.GutterHints { SupportedGutterHints: integer | nil }
+---@field hints? HintConfig
+---@field gutterHints? GutterHintConfig
+
+---@class (exact) Precognition.VirtLine
+---@field w PlaceLoc
+---@field e PlaceLoc
+---@field b PlaceLoc
+---@field Carat PlaceLoc
+---@field Dollar PlaceLoc
+
+---@class (exact) Precognition.GutterHints
+---@field G PlaceLoc
+---@field gg PlaceLoc
+---@field PrevParagraph PlaceLoc
+---@field NextParagraph PlaceLoc
+
+---@type HintConfig
+local defaultHintConfig = {
+    Carat = { text = "^", prio = 1 },
+    Dollar = { text = "$", prio = 1 },
+    w = { text = "w", prio = 10 },
+    b = { text = "b", prio = 9 },
+    e = { text = "e", prio = 8 },
+}
 
 ---@type Precognition.Config
 local default = {
     startVisible = true,
-    hints = {
-        ["^"] = { text = "^", prio = 1 },
-        ["$"] = { text = "$", prio = 1 },
-        ["w"] = { text = "w", prio = 10 },
-        -- ["W"] = "W",
-        ["b"] = { text = "b", prio = 9 },
-        ["e"] = { text = "e", prio = 8 },
-        -- ["ge"] = "ge", -- should we support multi-char / multi-byte hints?
-    },
+    showBlankVirtLine = true,
```

Comment:
imo this should default to false, I think that's more intuitive. Thoughts?

Reply:
I did think this, however, I'm thinking for a beginning. Which is probably who this plugin is primarily aimed at. Seeing a blank line implies there are no motions available, so my preference would be true by default